src/event-detail/ui/event-detail.controller.js

```diff
@@ -382,9 +382,15 @@ function EventDetail(
   function labelRemoved(label) {
     clearLabelsError();
 
-    offerLabeller
+    return $q(function(resolve, reject) {
```

Comment:
Should it also be done in the `place-detail.controller.js`? 
Maybe check also other places where labels can be added and removed :)